plugins/package-managers/gleam/src/main/kotlin/Gleam.kt

```diff
@@ -0,0 +1,661 @@
+            hexClient,
+            projectVcs,
+            projectHomepageUrl,
+            issues
+        )
+
+        val allResolved = resolvedDeps + resolvedDevDeps
+        val packages = allResolved.values.mapTo(mutableSetOf()) { it.toOrtPackage(hexClient) }
+
+        val scopes = buildSet {
+            if (resolvedDeps.isNotEmpty()) {
+                add(
+                    Scope(
+                        name = SCOPE_DEPENDENCIES,
+                        dependencies = resolvedDeps.values.map { PackageReference(it.toIdentifier()) }.toSet()
+                    )
+                )
+            }
+
+            if (resolvedDevDeps.isNotEmpty()) {
+                add(
+                    Scope(
+                        name = SCOPE_DEV_DEPENDENCIES,
+                        dependencies = resolvedDevDeps.values.map { PackageReference(it.toIdentifier()) }.toSet()
+                    )
+                )
+            }
+        }
+
+        issues += Issue(
+            source = projectType,
+            message = "Dependencies were resolved dynamically as no lockfile was present. " +
+                "Only the latest matching versions of direct dependencies were resolved without " +
+                "transitive dependency resolution. The results are not reproducible. " +
+                "Consider running 'gleam deps download' to generate a manifest.toml lockfile.",
+            severity = Severity.WARNING
+        )
+
+        val vcs = VcsHost.parseUrl(vcsUrl)
+        val project = createProject(definitionFile, gleamToml, scopes, vcs, projectVcs)
+        return listOf(ProjectAnalyzerResult(project, packages, issues))
+    }
+
+    private fun resolveHexDependency(
+        name: String,
+        requirement: String,
+        hexClient: HexApiClient,
+        issues: MutableList<Issue>
+    ): ResolvedPackage.Hex? {
+        val packageInfo = hexClient.getPackageInfo(name) ?: return null
+
+        // Get all versions from the releases list (Hex API returns them newest first)
+        val versions = packageInfo.releases.map { it.version }
+
+        // Find the latest version that matches the requirement
+        val version = findMatchingVersion(versions, requirement)
+        if (version == null) {
+            issues += Issue(
+                source = projectType,
+                message = "No version of '$name' satisfies requirement '$requirement'. Skipping.",
+                severity = Severity.WARNING
+            )
+            return null
+        }
+
+        val releaseInfo = hexClient.getReleaseInfo(name, version) ?: return null
+
+        return ResolvedPackage.Hex(
+            name = name,
+            version = version,
+            checksum = releaseInfo.checksum,
+            hexInfo = packageInfo
+        )
+    }
+
+    private fun resolveGitDependency(name: String, dep: GleamDependency.Git): ResolvedPackage.Git =
+        ResolvedPackage.Git(name = name, url = dep.url, ref = dep.ref)
```

Comment:
Nit: I guess using named arguments is not of too much value here, as the variables are mostly named the same.